### PR TITLE
Deprecate .link/1

### PR DESCRIPTION
**Pull request by mplatts**
https://github.com/petalframework/petal_components/issues/42

### CODE REVIEW

**Comment by mplatts** [comment]
Will merge when live view 0.18 is released

File: lib/petal_components/link.ex [hunk 1]
```diff
@@ -5,6 +5,11 @@ defmodule PetalComponents.Link do
   # prop label, :string
   # prop link_type, :string, options: ["a", "live_patch", "live_redirect", "button"]
   # slot default
+  @doc """
+  Deprecated in favour of the new phoenix_live_view link/1.
```
Comment (TwistingTwists) [inline lib/petal_components/link.ex:9]: can you please link to the documentation of 

`phoenix_live_view link/1 `
Can't seem to find this one in documentation. 

I am presently using link component though.

Comment (mplatts) [reply lib/petal_components/link.ex:9]: https://github.com/phoenixframework/phoenix_live_view/blob/master/CHANGELOG.md

Comment (TwistingTwists) [reply lib/petal_components/link.ex:9]: Thank you so much for this. :)

**Comment by krns** [comment]
Is there anything that needs to be done? I would like to help to use LiveView 0.18.0 im our projects :-)

**Comment by mplatts** [comment]
Live view 0.18 support is complete

**Comment by mplatts** [comment]
(see https://github.com/petalframework/petal_components/blob/main/UPGRADE_GUIDE.md)

**Pull request closed without merging**